frontend/src/components/InputText.js

```diff
@@ -17,21 +17,28 @@ class InputText extends Component {
 
     addItem(e) {
         if (this._inputElement.value !== "") {
+          const name = this._inputElement.value;
+          const nameCapitalized = name.charAt(0).toUpperCase() + name.slice(1).toLowerCase();
           var newItem = {
-            text: this._inputElement.value,
+            text: nameCapitalized,
             key: Date.now()
           };
-       
+                 
           this.setState((prevState) => {
+            prevState.items.forEach(function(item, index, object) {
+              if (item.text===newItem.text){
+                object.splice(index, 1);
```

Comment:
I don't get why you're adding the new item and then deleting it if it's a duplicate of an existing item.
Why not just skip adding the item if it exists? Consider something like this:

addItem(item) {
  // only add if the name doesn't exist in the list
  if(this.state.items.includes(capiltalizedName)){
    this.setState((state) => ({
      items: state.items.concat([capiltalizedName])
    }))
  }
}

Reply:
I tried this but because each item has a text and key component (needed to delete individual items), the includes function doesn't work